src/pc/lua/smlua_cobject.c

```
@@ -120,73 +120,187 @@ static const struct { enum LuaValueType lvt; const char *name; } sCustomFieldVal
+static int smlua_sort_lua_object_fields(const void *l, const void *r) {
+    const struct LuaObjectField *llof = (const struct LuaObjectField *) l;
+    const struct LuaObjectField *rlof = (const struct LuaObjectField *) r;
+    return strcmp(llof->key, rlof->key);
+}
+
+static struct LuaObjectTable *smlua_create_custom_object_table(struct Mod *mod) {
+    struct LuaObjectTable *ot = NULL;
+    u16 customFieldsCount = smlua_count_custom_fields(mod);
+    if (customFieldsCount > 0) {
+        ot = malloc(sizeof(struct LuaObjectTable));
+        ot->lot = LOT_OBJECT;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)